internal/cmd/gocdk/init_test.go

```diff
@@ -147,6 +148,83 @@ func TestInit(t *testing.T) {
 		}
 	})
 }
+func TestInferModulePath(t *testing.T) {
+	t.Run("NoGOPATHEntry", func(t *testing.T) {
+		dir, err := ioutil.TempDir("", testTempDirPrefix)
+		if err != nil {
+			t.Fatal(err)
+		}
+		defer func() {
+			if err := os.RemoveAll(dir); err != nil {
+				t.Error(err)
+			}
+		}()
+		ctx := context.Background()
+		pctx := &processContext{
+			workdir: dir,
+			env:     []string{},
+			stdin:   strings.NewReader(""),
+			stdout:  ioutil.Discard,
+			stderr:  ioutil.Discard,
+		}
+
+		fmt.Println(pctx.env)
+
+		const projectName = "myspecialproject"
+		if err := run(ctx, pctx, []string{"init", projectName}); err == nil {
+			t.Errorf("inferModulePath should error when --module-path is not" +
```

Comment:
Nit: you don't need to explain all this in the error; the name of the test provides most of the context. Just report what didn't meet expectations, e.g. `got nil error, want non-nil error`.